### PR TITLE
[#IOPID-581] io web profile function

### PR DESCRIPTION
**Pull request by arcogabbo**
This PR depends on #680 for a secret mentioned in the files

### List of changes
* added azure function regarding [io-web-profile-backend](https://github.com/pagopa/io-web-profile-backend) project

### Motivation and context
adding a new azure function

### Type of changes
- [x] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: src/domains/ioweb-app/06_function_ioweb_profile.tf [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+  }
+}
+
+module "function_ioweb_profile" {
+  source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//function_app?ref=v5.2.0"
+
+  resource_group_name = azurerm_resource_group.ioweb_profile_rg.name
+  name                = format("%s-ioweb-profile-fn", local.common_project)
+  location            = var.location
+  domain              = "IO-COMMONS"
```
[the remark this answers is not in the document]
Comment (arcogabbo) [reply src/domains/ioweb-app/06_function_ioweb_profile.tf]: applied in https://github.com/pagopa/io-infra/pull/678/commits/4abc4f0e61b95f2359f725bb2aac227cceb1b79f

File: src/domains/ioweb-app/99_variables.tf [hunk 2]
```diff
@@ -123,3 +123,38 @@ variable "tls_cert_check_helm" {
...
+  type        = string
+  description = "App service plan kind"
+  default     = null
+}
+
+variable "function_ioweb_profile_sku_size" {
+  type        = string
+  description = "App service plan sku size"
+  default     = null
+}
```
[the remark this answers is not in the document]
Comment (arcogabbo) [reply src/domains/ioweb-app/99_variables.tf]: sorry, I forgot to remove them. fixed in https://github.com/pagopa/io-infra/pull/678/commits/4abc4f0e61b95f2359f725bb2aac227cceb1b79f

File: src/domains/ioweb-app/06_function_ioweb_profile.tf [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    "blobs_retention_days"       = 0,
+  }
+
+  subnet_id = module.ioweb_profile_snet.id
+
+  allowed_subnets = [
+    module.ioweb_profile_snet.id,
+    data.azurerm_subnet.apim_snet.id,
+    data.azurerm_subnet.apim_v2_snet.id,
+  ]
```
[the remark this answers is not in the document]
Comment (arcogabbo) [reply src/domains/ioweb-app/06_function_ioweb_profile.tf]: nice catch. fixed in https://github.com/pagopa/io-infra/pull/678/commits/4abc4f0e61b95f2359f725bb2aac227cceb1b79f

File: src/domains/ioweb-app/06_function_ioweb_profile.tf [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+  )
+
+  subnet_id = module.ioweb_profile_snet.id
+
+  allowed_subnets = [
+    module.ioweb_profile_snet.id,
+    data.azurerm_subnet.azdoa_snet[0].id,
+    data.azurerm_subnet.apim_snet.id,
+    data.azurerm_subnet.apim_v2_snet.id,
+  ]
```
[the remark this answers is not in the document]
Comment (arcogabbo) [reply src/domains/ioweb-app/06_function_ioweb_profile.tf]: fixed in https://github.com/pagopa/io-infra/pull/678/commits/4abc4f0e61b95f2359f725bb2aac227cceb1b79f

File: src/domains/ioweb-app/06_function_ioweb_profile.tf [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+      // FF AND TESTERS
+      // --------------
+      FF_API_ENABLED = "BETA"
+      BETA_TESTERS   = data.azurerm_key_vault_secret.api_beta_testers.value
+
+      // ------------
+      // JWT Config
+      // ------------
+      BEARER_AUTH_HEADER       = "authorization"
+      EXCHANGE_JWT_ISSUER      = "api-web.io.pagopa.it/ioweb/auth"
```
Comment (gquadrati) [inline src/domains/ioweb-app/06_function_ioweb_profile.tf:54]: This token is issued by the function itself
```suggestion
      EXCHANGE_JWT_ISSUER      = "api-web.io.pagopa.it/ioweb/backend"
```

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: src/domains/ioweb-app/06_function_ioweb_profile.tf (hunk 5 above)
Comment (arcogabbo) [reply src/domains/ioweb-app/06_function_ioweb_profile.tf]: fixed in https://github.com/pagopa/io-infra/pull/678/commits/7d0365144e7607bc7d14a5e2add1fc3ce10d055d

File: src/domains/ioweb-app/01_network.tf [hunk 6]
```diff
@@ -51,3 +51,29 @@ data "azurerm_subnet" "private_endpoints_subnet" {
   virtual_network_name = local.vnet_common_name
   resource_group_name  = local.vnet_common_resource_group_name
 }
+
+data "azurerm_subnet" "apim_snet" {
+  name                 = "apimapi"
+  virtual_network_name = local.vnet_common_name
+  resource_group_name  = local.vnet_common_resource_group_name
+}
```
[the remark this answers is not in the document]
Comment (arcogabbo) [reply src/domains/ioweb-app/01_network.tf]: fixed in https://github.com/pagopa/io-infra/pull/678/commits/7d0365144e7607bc7d14a5e2add1fc3ce10d055d

**Review by gquadrati — DISMISSED** [review]

File: src/domains/ioweb-app/02_security.tf [hunk 7]
```diff
@@ -2,3 +2,41 @@ data "azurerm_key_vault" "kv" {
...
+#######
+resource "tls_private_key" "ioweb_profile_jwe_key" {
+  algorithm   = "ECDSA"
+  ecdsa_curve = "P256"
+}
+
+resource "tls_private_key" "ioweb_profile_jwt_key" {
+  algorithm = "RSA"
+  rsa_bits  = 2048
+}
```
[the remark this answers is not in the document]
Comment (arcogabbo) [reply src/domains/ioweb-app/02_security.tf:17]: this key generation has been used also in spid-login for ioweb

[the remark this answers is not in the document]
Comment (arcogabbo) [reply src/domains/ioweb-app/02_security.tf:17]: it'is also used on our modules(for example in [jwt_keys](https://github.com/pagopa/terraform-azurerm-v3/tree/main/jwt_keys)) that has been already used in other domains.

**Pull request merged**